pkg/monitor/cluster/certificateexpirationstatuses_test.go

```diff
@@ -0,0 +1,141 @@
+package cluster
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+	utiltls "github.com/Azure/ARO-RP/pkg/util/tls"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+func TestEmitCertificateExpirationStatuses(t *testing.T) {
+	expiration := time.Now().Add(time.Hour * 24 * 5)
+	expirationString := expiration.UTC().Format(time.RFC3339)
+	for _, tt := range []struct {
+		name            string
+		isManaged       bool
+		wantExpirations []map[string]string
+	}{
+		{
+			name:      "unmanaged domain",
+			isManaged: false,
+			wantExpirations: []map[string]string{
+				{
+					"subject":        "geneva.certificate",
+					"expirationDate": expirationString,
+				},
+			},
+		},
+		{
+			name:      "managed domain",
+			isManaged: true,
+			wantExpirations: []map[string]string{
+				{
+					"subject":        "geneva.certificate",
+					"expirationDate": expirationString,
+				},
+				{
+					"subject":        "contoso.aroapp.io",
+					"expirationDate": expirationString,
+				},
+				{
+					"subject":        "api.contoso.aroapp.io",
+					"expirationDate": expirationString,
+				},
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+
+			var secrets []runtime.Object
+			_, genevaCert, err := utiltls.GenerateTestKeyAndCertificate("geneva.certificate", nil, nil, false, false, func(template *x509.Certificate) {
+				template.NotAfter = expiration
+			})
```

Comment:
I think it is a good idea to remove the inline function definition. This will have benefits in the long run. Something like:
replace above chunk with....
```go
tweakTemplateFn := tweakTemplateFn(expiration)
_, genevaCert, err := utiltls.GenerateTestKeyAndCertificate("geneva.certificate", nil, nil, false, false, tweakTemplateFn)
```
and create
```go
func tweakTemplateFn(expiration time.Time) func(*x509.Certificate) {
	return func(template *x509.Certificate) {
		template.NotAfter = expiration
	}
}
```
Also, we can now reuse the function as it was originally hardcoded in two different places.